frost-secp256k1/src/lib.rs

```diff
@@ -0,0 +1,396 @@
+#![allow(non_snake_case)]
+#![deny(missing_docs)]
+#![doc = include_str!("../README.md")]
+
+use digest::generic_array::GenericArray;
+use k256::{
+    elliptic_curve::{
+        bigint::{Encoding, U384},
+        group::prime::PrimeCurveAffine,
+        hash2curve::{ExpandMsg, ExpandMsgXmd, Expander},
+        sec1::{FromEncodedPoint, ToEncodedPoint},
+        Field as FFField, PrimeField,
+    },
+    AffinePoint, ProjectivePoint, Scalar,
+};
+use rand_core::{CryptoRng, RngCore};
+use sha2::{digest::Update, Digest, Sha256};
+
+use frost_core::{frost, Ciphersuite, Field, Group};
+
+#[cfg(test)]
+mod tests;
+
+pub use frost_core::Error;
+
+#[derive(Clone, Copy)]
+/// An implementation of the FROST(Ed25519, SHA-512) ciphersuite scalar field.
```

Comment:
I'm never sure where the right place is but I think the `#derive[]` should go closest to the declaration and the rustdoc just above it

Reply:
This way it's a bit simpler for `gendoc` to handle but it can be worked around, what do you think?

Reply:
Rust std lib tends to put annotations close to the declaration and the rustdoc above it:

https://doc.rust-lang.org/src/core/error.rs.html

<img width="546" alt="image" src="https://user-images.githubusercontent.com/552961/201231782-da8112f2-e3f1-43fe-af69-dadb0d1866ca.png">